packages/http/src/client.ts

```diff
@@ -0,0 +1,131 @@
+import { IPrismOutput } from '@stoplight/prism-core';
+// @ts-ignore
+import logger from 'abstract-logging';
+import { getStatusText } from 'http-status-codes';
+import { defaults } from 'lodash';
+import { parse as parseQueryString } from 'querystring';
+import { parse as parseUrl } from 'url';
+import { createInstance } from '.';
+import { forwarder } from './forwarder';
+import { getHttpOperationsFromFile } from './getHttpOperations';
+import { mocker } from './mocker';
+import { router } from './router';
+import { IHttpConfig, IHttpRequest, IHttpResponse, IHttpUrl } from './types';
+import { validator } from './validator';
+
+const createNewClientInstance = async (defaultConfig: IHttpConfig, spec: string): Promise<PrismHttp> => {
+  const obj = createInstance(defaultConfig, {
+    logger,
+    router,
+    forwarder,
+    validator,
+    mocker,
+  });
+
+  const resources = await getHttpOperationsFromFile(spec);
+
+  const request: RequestFunction = async (url, input, config) => {
+    const parsedUrl = parseUrl(url);
+
+    if (!parsedUrl.pathname) throw new Error('path name must alwasy be specified');
+
+    const httpUrl: IHttpUrl = {
+      baseUrl: parsedUrl.host ? `${parsedUrl.protocol}//${parsedUrl.host}` : undefined,
+      path: parsedUrl.pathname,
+      query: parseQueryString(parsedUrl.query || ''),
+    };
+
+    const data = await obj.request(
+      {
+        ...input,
+        url: httpUrl,
+      },
+      resources,
+      config,
+    );
+
+    const o: PrismOutput = {
+      status: data.output.statusCode,
+      statusText: getStatusText(data.output.statusCode),
```

Comment:
Can we see if there is an actual status text being returned? People can override them so `200 Monkeys` Is actually perfectly valid. We’d want to see statusText as Monkeys and this code would make it “OK”

Reply:
Good point. Currently when Prism is mocking, basically there's no status text returned. That's why I used this function. However in case we're hitting a real server it actually make sense to just return what the server returned back to us. I'll look into this.